src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -131,8 +137,19 @@ def save_as_table(
                 "Unsupported table type. Expected table types: temp/temporary, transient"
             )
 
+        if column_order:
```

Comment:
Given that the default value is `index`, I don't think we should allow `None`.

Reply:
User can still possibly pass `None` to override the default value.

Reply:
Yes, and I think we should fail in that case.

Reply:
+1

Reply:
There is no harm to allow `None`/`""` then I prefer allowing it.
It's possible that someday we need it to be `None` because another API can set `column_order`, like for `mode`.

Reply:
I feel strongly against this. If another API can set `column_order`, that API should obey the input limitation set by this API, where the legit inputs are `index` and `name`.

Reply:
As per offline discussion, I updated the input validation to allow only "index" and "name".